### PR TITLE
Fix klog error in `pkg/proxy`

### LINKED ISSUE #105148: Fix incorrectly migrated structured logs
#### What happened:
When working on enabling static analysis for structured logs on whole codebase https://github.com/kubernetes/kubernetes/issues/102439 we have noticed that there are multiple places that were incorrectly migrated. https://github.com/kubernetes/kubernetes/pull/103293#issuecomment-922694019
Before we enable static analysis we need to fix those places.

#### What you expected to happen:
Running logcheck should not detect any invalid log calls.

#### How to reproduce it (as minimally and precisely as possible):
Run klog logcheck script on kubernetes repo with --allow-unstructured flag. https://github.com/kubernetes/klog/tree/main/hack/tools/logcheck

#### Anything else we need to know?:
Please use klog from master branch as we just fixed Exit* kubernetes/klog#264

/wg structured-logging

### PR DESCRIPTION
**Pull request by jyz0309**
Signed-off-by: jyz0309 <45495947@qq.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
According to this [comment](https://github.com/kubernetes/kubernetes/pull/105144#discussion_r712028012), some logger helper function have been called error. So remove them.
#### Which issue(s) this PR fixes:

Fixes #105148 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/proxy/winkernel/proxier.go [hunk 1]
```diff
@@ -841,7 +830,9 @@ func deleteAllHnsLoadBalancerPolicy() {
 		return
 	}
 	for _, plist := range plists {
-		LogJson("policyList", plist, "Remove Policy", 3)
+		if jsonString, err := json.Marshal(plist); err == nil {
+			klog.V(3).InfoS("Remove policy", "policyList", jsonString)
```
Comment (jyz0309) [inline pkg/proxy/winkernel/proxier.go:834]: FYI, `InfoS` already handle []byte display in [klog/pull/237](https://github.com/kubernetes/klog/pull/237). So we don't need to string().

**Comment by shivanshu1333** [comment]
Nicely done @jyz0309, thanks!
lgtm

File: pkg/proxy/winkernel/proxier.go [hunk 2]
```diff
@@ -1209,7 +1200,9 @@ func (proxier *Proxier) syncProxyRules() {
 			}
 
 			// Save the hnsId for reference
-			LogJson("endpointInfo", newHnsEndpoint, "Hns Endpoint resource", 1)
+			if jsonString, err := json.Marshal(newHnsEndpoint); err == nil {
```
Comment (serathius) [inline pkg/proxy/winkernel/proxier.go:1203]: ditto

File: pkg/proxy/winkernel/proxier.go [hunk 3]
```diff
@@ -454,7 +443,7 @@ func newSourceVIP(hns HostNetworkService, network string, ip string, mac string,
 }
 
 func (ep *endpointsInfo) Cleanup() {
-	Log(ep, "Endpoint Cleanup", 3)
+	klog.V(3).InfoS("Endpoint cleanup", "spewConfig", spewSdump(ep))
```
Comment (serathius) [inline pkg/proxy/winkernel/proxier.go:446]: What spewSdump function does? We should always prefer to pass original struct and depend on logging library sanitization.

Comment (jyz0309) [reply pkg/proxy/winkernel/proxier.go:446]:
```
func spewSdump(v interface{}) string {
	scs := spew.NewDefaultConfig()
	scs.DisableMethods = true
	return scs.Sdump(v)
}
```
I don't think logging lib can handle this... Maybe I can move this logic to `Cleanup` function?😳

Comment (jyz0309) [reply pkg/proxy/winkernel/proxier.go:446]: Or just show the `endpointsInfo`😳

Comment (serathius) [reply pkg/proxy/winkernel/proxier.go:445]: This code still doesn't show what spew does. Does it just marshall? Can you provide comparison of log that use spew and if we just skip it?

Comment (jyz0309) [reply pkg/proxy/winkernel/proxier.go:445]: A small test:
```
klog.InfoS("Endpoint cleanup", "spewConfig", spewSdump(p))
***the log result:***
I0921 21:10:52.783833   45175 main.go:51] "Endpoint cleanup" spewConfig="(*main.endpointsInfo)(0xc0000be2a0)({\n ip: (string) (len=9) \"127.0.0.1\",\n port: (uint16) 3307,\n isLocal: (bool) false,\n macAddress: (string) (len=5) \"A1:B2\",\n hnsID: (string) (len=3) \"333\",\n refCount: (*uint16)(<nil>),\n providerAddress: (string) (len=7) \"0.0.0.0\",\n ready: (bool) false,\n serving: (bool) false,\n terminating: (bool) false\n})\n"

klog.InfoS("Endpoint cleanup", "spewConfig", p)
***the log result:***
I0921 21:10:52.784251   45175 main.go:52] "Endpoint cleanup" spewConfig=&{ip:127.0.0.1 port:3307 isLocal:false macAddress:A1:B2 hnsID:333 refCount:<nil> providerAddress:0.0.0.0 ready:false serving:false terminating:false}
```
Well, it seems that just a little difference.🙈

Comment (serathius) [reply pkg/proxy/winkernel/proxier.go:445]: In that case, please remove spew

Comment (jyz0309) [reply pkg/proxy/winkernel/proxier.go:445]: I see that `proxier.go` have other klog call `spewSdump(v interface{})`, should I remove them all?

Comment (serathius) [reply pkg/proxy/winkernel/proxier.go]: That would be preferred.

**Comment by shivanshu1333** [comment]
Kindly change releated to #105148 to Fixes #105148, as this using "Fixes" will close the issue once the PR gets merged, this PR will solve the original issue completely.
Thanks!

File: pkg/proxy/winkernel/proxier.go [hunk 4]
```diff
@@ -1236,7 +1218,7 @@ func (proxier *Proxier) syncProxyRules() {
 			continue
 		}
 
-		klog.V(4).Infof("Trying to Apply Policies for service", "spewConfig", spewSdump(svcInfo))
+		klog.V(4).Infof("Trying to Apply Policies for service", "serviceInfo", svcInfo)
```
Comment (serathius) [inline pkg/proxy/winkernel/proxier.go:1221]:
```suggestion
		klog.V(4).Infof("Trying to apply policies for service", "serviceInfo", svcInfo)
```

File: pkg/proxy/winkernel/hnsV2.go [hunk 5]
```diff
@@ -202,7 +202,7 @@ func (hns hnsV2) getLoadBalancer(endpoints []endpointsInfo, flags loadBalancerFl
 			} else if len(plist.FrontendVIPs) != 0 {
 				continue
 			}
-			LogJson("policyList", plist, "Found existing Hns loadbalancer policy resource", 1)
+			klog.V(1).InfoS("found existing Hns loadbalancer policy resource", "policyList", plist)
```
Comment (serathius) [inline pkg/proxy/winkernel/hnsV2.go:205]:
```suggestion
			klog.V(1).InfoS("Found existing Hns loadbalancer policy resource", "policyList", plist)
```

File: pkg/proxy/winkernel/hnsV1.go [hunk 6]
```diff
@@ -194,7 +194,7 @@ func (hns hnsV1) getLoadBalancer(endpoints []endpointsInfo, flags loadBalancerFl
 			} else if len(elbPolicy.VIPs) != 0 {
 				continue
 			}
-			LogJson("policyList", plist, "Found existing Hns loadbalancer policy resource", 1)
+			klog.V(1).InfoS("found existing Hns loadbalancer policy resource", "policyList", plist)
```
Comment (serathius) [inline pkg/proxy/winkernel/hnsV1.go:197]: User guide specifies that "policies" is better "policyList".
Can you fix it in other PR cases?

File: pkg/proxy/winkernel/hnsV2.go [hunk 7]
```diff
@@ -281,8 +281,7 @@ func (hns hnsV2) getLoadBalancer(endpoints []endpointsInfo, flags loadBalancerFl
 		return nil, err
 	}
 
-	LogJson("hostComputeLoadBalancer", lb, "Hns loadbalancer policy resource", 1)
-
+	klog.V(1).InfoS("Hns loadbalancer policy resource", "hostComputeLoadBalancer", lb)
```
Comment (serathius) [inline pkg/proxy/winkernel/hnsV2.go:284]:
```suggestion
	klog.V(1).InfoS("Hns loadbalancer policy resource", "loadBalancer", lb)
```

File: pkg/proxy/winkernel/proxier.go [hunk 8]
```diff
@@ -1167,7 +1148,7 @@ func (proxier *Proxier) syncProxyRules() {
 
 					newHnsEndpoint, err = hns.createEndpoint(hnsEndpoint, hnsNetworkName)
 					if err != nil {
-						klog.ErrorS(err, "Remote endpoint creation failed", "spewConfig", spewSdump(hnsEndpoint))
+						klog.ErrorS(err, "Remote endpoint creation failed", "endpointInfo", hnsEndpoint)
```
Comment (serathius) [inline pkg/proxy/winkernel/proxier.go:1151]:
```suggestion
						klog.ErrorS(err, "Remote endpoint creation failed", "endpointsInfo", hnsEndpoint)
```

**Comment by serathius** [comment]
Hey @jyz0309, I have already lgtmed the PR, you just need approval now. Please ping @sbangari or find alternative approver. On https://kubernetes.slack.com/archives/C020CCMUEAX/p1631887842073600 I have discussed how to do that.

**Comment by jyz0309** [comment]
> Hey @jyz0309, I have already lgtmed the PR, you just need approval now. Please ping @sbangari or find alternative approver. On https://kubernetes.slack.com/archives/C020CCMUEAX/p1631887842073600 I have discussed how to do that.

Oh, I will do it. Thanks for your help! You help me a lot!

**Comment by jyz0309** [comment]
It seems that need the `lgtm` and `approve` labels after I rebase this PR😕

**Comment by jyz0309** [comment]
PTAL  @sbangari

**Comment by jyz0309** [comment]
PTAL @ksubrmnn @sbangari

**Review by ksubrmnn — APPROVED** [review]

**Pull request merged**